src/shared/Core/UriExtensions.cs

```diff
@@ -102,6 +102,16 @@ public static IEnumerable<string> GetGitConfigurationScopes(this Uri uri)
                 }
             }
 
+            // check whether the host specification contains of hostname only
+            // this usually means, the git hosting instance is part of your local
+            // network.
+            if (!(string.IsNullOrWhiteSpace(host) || string.IsNullOrEmpty(host)) &&
+                !host.Contains("."))
+            {
+                yield return $"{schemeAndDelim}{host}";
+                yield break;
```

Comment:
Can you clarify why we're using `yield break`? Something like:

```suggestion
                // If we have reached this point, there are no more subdomains to unfold, so exit early.
                yield break;
```